students/Ramkumar_Rajanbabu/lesson09/donor_models.py

```diff
@@ -0,0 +1,160 @@
+#Main Script
+
+import sys
+
+class Donor():
+    """Single donor information"""
+    
+    def __init__(self, name, donations = None):
+        self._name = name
+        if donations is not None:
+            self._donations = donations
+        else: 
+            self._donations = []
+    
+    def __repr__(self):
+        return f"Donor: {self.name}, Donations: {self.donations}"
+    
+    def add_donations(self, amount):
+        self.donations.append(amount)
+    
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def donations(self):
+        return self._donations
+    
+    @property
+    def total_donations(self):
+        return sum(self._donations)
+    
+    @property
+    def number_donations(self):
+        return len(self._donations)
+    
+    @property
+    def average_donations(self):
+        return int(self.total_donations/self.number_donations)
+
+class DonorCollection():
+    """All donor information"""
+    
+    def __init__(self, donors=None):
+        if donors is not None:
+            self.donors = donors
+        else:
+            self.donors = []
+    
+    def add_new_donor(self, donor):
+        self.donors.append(donor)
+    
+    def list_donor(self):
+        return [donor.name for donor in self.donors]
+    
+    def list_donations(self):
+        return [donor.donations for donor in self.donors]
+    
+    def send_thanks(self):
+        name = None
+        while not name:
+            name = input_name()
+            if name == "list":
+                print(self.list_donor())
+                name = None
+        
+        donation_amount = None
+        while not donation_amount:
+            #Exception handling
+            try:
+                donation_amount = input_donation()
+            except ValueError: #ValueError if not integer/float
+                print("Please enter an integer.")
```

Comment:
there should be no input functions in your data classes (even if input() is not directly called)